packages/repository-json-schema/src/build-schema.ts

```diff
@@ -142,16 +151,25 @@ export function metaToJsonProperty(meta: PropertyDefinition): JSONSchema {
  * reflection API
  * @param ctor Constructor of class to convert from
  */
-export function modelToJsonSchema(ctor: Function): JSONSchema {
+export function modelToJsonSchema(
+  ctor: Function,
+  options: JsonSchemaOptions = {},
+): JSONSchema {
+  options.visited = options.visited || {};
```

Comment:
Please don't modify original `options`, see the discussion above.

Reply:
applied and new test case added.